src/app/_components/PageHeader.tsx

```diff
@@ -39,43 +34,40 @@ function Title({ children }: TitleProps) {
   )
 }
 
-function PageHeaderImage({ image }: Pick<PageHeaderProps, 'image'>) {
-  const { type, ...rest } = image
+const sharedContainerStyle = 'relative aspect-video lg:aspect-auto lg:w-1/2'
+const sharedImageStyle = 'h-full w-full rounded-lg border border-brand-100'
+const imageSizes = buildImageSizeProp({ startSize: '100vw', md: '660px' })
 
-  if (type === 'static') {
-    const staticImage = rest as StaticImageProps
+function PageHeaderImage({ image }: Pick<PageHeaderProps, 'image'>) {
+  const isDynamicImage = 'src' in image
+  const isStaticImage = 'data' in image
 
+  if (isStaticImage) {
     return (
       <div className={sharedContainerStyle}>
         <StaticImage
-          {...staticImage}
+          {...image}
           fill
           priority
           quality={100}
           className={sharedImageStyle}
-          sizes={buildImageSizeProp({ startSize: '100vw', lg: '480px' })}
+          sizes={imageSizes}
         />
       </div>
     )
   }
 
-  if (type === 'dynamic') {
-    const dynamicImage = rest as DynamicImageProps
-
+  if (isDynamicImage) {
     return (
       <div className={sharedContainerStyle}>
         <DynamicImage
-          {...dynamicImage}
+          {...image}
           fill
           priority
           quality={100}
           className={sharedImageStyle}
-          sizes={buildImageSizeProp({ startSize: '100vw', lg: '480px' })}
-          fallback={{
-            ...dynamicImage.fallback,
-            className: sharedImageStyle,
-            fill: true,
-          }}
+          sizes={imageSizes}
```

Comment:
I like this approach with `sizes`, wondering if we could do that we others so instead of` buildImageSizeProp({ startSize: '100vw', md: '480px' })`, we have `imageMediumSizes` or something like that, and we have functions centralised in case we need to change.